pageserver/src/tenant.rs

```diff
@@ -1226,8 +1226,20 @@ impl Tenant {
             "Cannot create timelines on inactive tenant"
         );
 
-        if self.get_timeline(new_timeline_id, false).is_ok() {
+        if let Ok(existing) = self.get_timeline(new_timeline_id, false) {
             debug!("timeline {new_timeline_id} already exists");
+
+            if let Some(remote_client) = existing.remote_client.as_ref() {
+                // wait for uploads to complete even if this is a random old/very active timeline
```

Comment:
> console uses pre-generated timeline ids in create timeline/tenant requests.

It does, because otherwise retries could be painful. Now we retry with the same id for tenant/timeline creation

Reply:
Thanks!